fastscanner/benchmarks/benchmark_redis_read.py

```diff
@@ -0,0 +1,155 @@
+import asyncio
+import json
+import logging
+import os
+import time
+from datetime import datetime
+
+import redis.asyncio as aioredis
+
+from fastscanner.pkg import config
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+SYMBOLS_FILE = "data/symbols/symbols.json"
+STREAM_PREFIX = "candles_min_"
+NO_DATA_TIMEOUT = 10
+LAST_ID_KEY = "stream_processor:last_ids"
+
+batch_start_time = None
+last_received_time = None
+batch_lock = asyncio.Lock()
+total_messages = 0
+
+last_ids: dict[str, str] = {}
+
+
+async def get_or_load_symbols(redis: aioredis.Redis) -> list[str]:
+    os.makedirs(os.path.dirname(SYMBOLS_FILE), exist_ok=True)
+
+    if os.path.exists(SYMBOLS_FILE):
+        with open(SYMBOLS_FILE, "r") as f:
+            symbols = json.load(f)
+            logger.info(f"Loaded {len(symbols)} symbols from file.")
+            return symbols
+
+    symbols = []
+    async for key in redis.scan_iter(f"{STREAM_PREFIX}*"):
+        if await redis.type(key) == "stream":
+            symbol = key.replace(STREAM_PREFIX, "")
+            symbols.append(symbol)
+
+    with open(SYMBOLS_FILE, "w") as f:
+        json.dump(symbols, f)
+
+    logger.info(f"Discovered and saved {len(symbols)} symbols.")
+    return symbols
+
+
+async def get_last_id(redis: aioredis.Redis, stream_key: str) -> str:
+    last_id = await redis.hget(LAST_ID_KEY, stream_key)
+    if last_id is not None and isinstance(last_id, str):
+        return last_id
+
+    last_entry = await redis.xrevrange(stream_key, count=1)
+    return last_entry[0][0] if last_entry else "0-0"
+
+
+async def save_last_id(redis: aioredis.Redis, stream_key: str, last_id: str) -> None:
+    await redis.hset(LAST_ID_KEY, stream_key, last_id)
```

Comment:
This is gonna increase the latency. Also xrevrange is a O(1) method so there is no point on caching the result of an O(1) method.

We don't really need the cache logic for the last_id. Just store it in a global variable in a dictionary. Or in a class variable when you implement the subscribe method for the RedisChannel.
```
symbol_to_last_id: dict[str, str] = {} 
```